### PR TITLE
Road to 1.0, plus SDL conversion

### PR DESCRIPTION
**Pull request by pedro-w**
There are sure to be some bugs/missing features but the basics of it are working (tested on MacOS)
Please check it out and review!

### CODE REVIEW

**Comment by z9484** [comment]
Nice work. Those last commits did the trick. Nice thing with SDL, I'm pretty sure we should be able to access some better scaling options without too much of a hassle.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,44 +1,35 @@
...
+include(FindPkgConfig)
 
 project(kq-fork)
 
-set (CMAKE_MODULE_PATH "${CMAKE_MODULE_PATH};${CMAKE_CURRENT_SOURCE_DIR};${CMAKE_CURRENT_SOURCE_DIR}/maps")
+set (CMAKE_MODULE_PATH "${CMAKE_MODULE_PATH}"
+  "${CMAKE_CURRENT_SOURCE_DIR}")
 
-find_package(Allegro REQUIRED)
+find_package(SDL2 CONFIG REQUIRED)
```
Comment (OnlineCop) [inline CMakeLists.txt:9]: Where can I find the SDL.cmake file that would go with this? Or, how do I generate whichever file it's needing?

**Comment by OnlineCop** [comment]
I think I'm confusing myself more than anything now.

* I see an `SDL-plus-1.0` branch. I can't remember whether that was supposed to be merged in, or what's going on with that one.
* I saw the `road_to_1.0` PR from @pedro-w. I merged that one in. Or was I supposed to have rebased that in?

Does `SDL-plus-1.0` exist anymore? Or, does that become our current `road_to_1.0`?

File: CMakeLists.txt (hunk 1 above)
Comment (pedro-w) [reply CMakeLists.txt:9]: What platform are you using? The relevant package is part of SDL and should have been installed by your system's package manager.

Comment (OnlineCop) [reply CMakeLists.txt:9]: Windows 10. The sdl install instructions only say to extract the files somewhere, but nothing concrete.

I miss Linux.

**Comment by pedro-w** [comment]
I believe you've done the right thing - merged pedro-w's SDL-plus-1.0 into onlinecop's road_to_1.0. The former can be deleted (by me) because it doesn't now have anything that's not in your repository.

**Comment by pedro-w** [comment]
> Nice work. Those last commits did the trick. Nice thing with SDL, I'm pretty sure we should be able to access some better scaling options without too much of a hassle.

Right - all the graphics are drawn onto one 320x240x8bit pseudo-bitmap which is then rendered to the screen in one go. Should be simple to just fit it to a user-resizable window. Or are you thinking of upscaling algorithms e.g. 2xSAI ?

**Comment by OnlineCop** [comment]
For any of you that are getting this working on Windows, would you mind giving me a hand? I'm trying this with Visual Studio 2019, but am open to pretty much anything that will work right now, so I can start coding.

1. We can continue the discussion in this (merged) PR thread.
2. We can start a conversation elsewhere, such as to another thread, or to the [wiki](https://github.com/OnlineCop/kq-fork/wiki) (or one of its subpages: [EngineRework](https://github.com/OnlineCop/kq-fork/wiki/EngineRework) or [Road to 1.0](https://github.com/OnlineCop/kq-fork/wiki/Road-to-1.0))
3. On [Libera IRC](https://web.libera.chat/?nick=kqfork-?&chan=#kqfork) (this link provided for convenience if you just want to log in anonymously; I've only had to log in when trying to connect via a VPN) in the **#kqfork** channel.
4. Discord? Twitter? Slack? Google Hangouts-or-whatever-they-have-now? Teams?

**Comment by pedro-w** [comment]
We might struggle with real-time chat options since we are all busy and in different time zones (?). If I can find time today I'll scratch my windows folder and start again, making notes. Or Z9484 could just have the answer you seek!

**Comment by z9484** [comment]
I added the steps I took https://github.com/OnlineCop/kq-fork/wiki/EngineRework#building-on-windows-with-new-sdl-branch-on-road_to_10
Its probably more involved then it needs to be but it looks like I need to learn more about vcpkg first.

For scaling I was really meaning dynamic windows sizing. Pygame has a SCALED flag but looks like it must be a pygame thing and not built in to SDL. 

But we could add filters like 2xSAI or others for people who like them.

**Comment by z9484** [comment]
Hmm looks like running (holding down ctrl) is broken for some reason now.

**Comment by pedro-w** [comment]
Nice one. Unfortunately not 100% there yet on CMake because it now doesn't work on Debian, but I know how to fix it. First priority is to get everyone able to compile and run it, then we can get back to fixing bugs and finishing what needs to be finished!

ps. can you file an issue for the running - I'm not surprised it's broken; I had to rework the timing quite a bit to get away from Allegro 4, where it ran internally at 100Hz and updated the screen whenever it felt like it.

**Comment by OnlineCop** [comment]
It looks like I can now compile with this PR (see latest notes in #78), but I get a runtime error within VS2022: when KQ_DATADIR is set to `.`, it does not correctly locate that directory and causes the game to fail when loading `Misc.png`.

If I set it to `${CMAKE_CURRENT_SOURCE_DIR}` instead, it runs and debugs fine within Visual Studio.

    -set(KQ_DATADIR "." CACHE PATH "Where the data dirs are going to be installed")
    +set(KQ_DATADIR "${CMAKE_CURRENT_SOURCE_DIR}" CACHE PATH "Where the data dirs are going to be installed")

What are your thoughts? Once it's released, it obviously makes sense to be relative to its installation directory, but for now, would it make sense to be the same as the project root?

**Comment by pedro-w** [comment]
That seems to be a sensible default - I think we should do that

**Pull request merged**